### PR TITLE
Require newlines

### PR DESCRIPTION
**Pull request by JesseStephenBrown**
This will check for mandatory newline at end, add an EOF token to the end of the tokens list, and fixed some issues with testprogs not having newline and not being spaces instead of tabs

### CODE REVIEW

File: testProgs/medium.mp [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+  writeln ('Number 2 = ', second);
+  writeln ('Number 3 = ', third);
+  writeln ('Number 4 = ', fourth);
+  writeln ('Number 5 = ', fifth);
+
+  writeln ('Sum = ', sum);
+
+  average := sum / 5;
+  writeln ('Average = ', average);
+end.
```
Comment (brianpmaher) [inline testProgs/medium.mp:39]: This change was made in my PR as well

File: testProgs/parseError.mp [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+  this__issupposed to not work : real;
 
 begin
-	writeln ('Variable: ', variable);
-	
-end. 
\ No newline at end of file
+  writeln ('Variable: ', variable);
+
+end.
```
Comment (brianpmaher) [inline testProgs/parseError.mp:21]: ditto

File: testProgs/scanError.mp [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 begin
-	(* This is supposed to be an error *)
-	writeln ('Hello there, error?')
-	
-end. 
\ No newline at end of file
+  (* This is supposed to be an error *)
+  writeln ('Hello there, error?')
+
+end.
```
Comment (brianpmaher) [inline testProgs/scanError.mp:22]: ditto

File: testProgs/simple.mp [hunk 4]
```diff
@@ -1,4 +1,4 @@
 program Hello;
   begin
   writeln ('Hello, world.')
-end.
\ No newline at end of file
+end.
```
Comment (brianpmaher) [inline testProgs/simple.mp:4]: ditto_ag_anime

**Comment by brianpmaher** [comment]
+1

**Comment by JesseStephenBrown** [comment]
Lol, that's a big ditto. I'll handle the merge conflicts.

**Comment by seanrogers2657** [comment]
Yay! Merge conflicts! +1

**Comment by brianpmaher** [comment]
+1 again after merge conflicts

**Pull request merged**